timelock-impl/src/main/java/com/palantir/atlasdb/timelock/batch/MultiClientConjureTimelockResource.java

```diff
@@ -212,6 +266,34 @@ private ListenableFuture<Map.Entry<Namespace, LeaderTime>> getNamespacedLeaderTi
                 MoreExecutors.directExecutor());
     }
 
+    private <K, T, R> ListenableFuture<R> transformSetAsync(
+            Set<K> input, Function<K, ListenableFuture<T>> function, Function<Map<K, T>, R> mapFunction) {
+        List<ListenableFuture<Entry<K, T>>> entryFutures = input.stream()
+                .map(element -> Futures.transform(
+                        function.apply(element),
+                        result -> Maps.immutableEntry(element, result),
+                        MoreExecutors.directExecutor()))
+                .collect(Collectors.toList());
+        return Futures.transform(
+                Futures.allAsList(entryFutures),
+                entries -> mapFunction.apply(ImmutableMap.copyOf(entries)),
+                MoreExecutors.directExecutor());
+    }
```

Comment:
I can write this in terms of transformMapAsync but preferred to avoid the waste.

Also, would love recommendation of some third party API that does something similar to what I did here